frontend/src/components/pages/VolunteerDashboard/ShiftCard.tsx

```diff
@@ -0,0 +1,129 @@
+import { Box, Button, Stack, Text } from "@chakra-ui/react";
+import { format, parse } from "date-fns";
+import React, { useEffect, useState } from "react";
+import { useHistory } from "react-router-dom";
+
+import DonorAPIClient from "../../../APIClients/DonorAPIClient";
+import * as Routes from "../../../constants/Routes";
+import {
+  CheckInWithShiftType,
+  ScheduleWithShiftType,
+  ShiftType,
+} from "../../../types/VolunteerTypes";
+import CardField from "../../common/CardField";
+import { getShiftColor } from "./types";
+
+interface CheckInOrScheduleProps {
+  donorId?: string;
+  isPickup?: boolean;
+  pickupLocation?: string;
+  volunteerTime?: string;
+  startTime?: string;
+  startDate?: string;
+  notes: string;
+  type: ShiftType;
+}
+const VolunteerShiftCard = ({
+  shift,
+}: {
+  shift: CheckInWithShiftType | ScheduleWithShiftType;
+}): JSX.Element => {
+  const {
+    donorId,
+    isPickup,
+    pickupLocation,
+    volunteerTime,
+    startTime,
+    startDate,
+    notes,
+    type,
+  } = shift as CheckInOrScheduleProps;
+  const [businessName, setBusinessName] = useState<string>("");
+
+  const history = useHistory();
+
+  const dateLocal = () => {
+    if (startDate) {
+      return format(new Date(startDate), "EEE MMM dd, yyyy");
+    }
+    return startTime && format(new Date(startTime), "EEE MMM dd, yyyy");
+  };
+
+  const timeLocal = () => {
+    if (startDate) {
+      return format(new Date(startDate), "h:mm aa");
+    }
+
+    return (
+      volunteerTime &&
+      format(parse(volunteerTime, "kk:mm", new Date()), "h:mm aa")
```

Comment:
I'm not sure `volunteerTime` here is always in `kk:mm` format?
I tried logging it and got different values
```
2021-11-06 19:25:32.806 +00:00
8:00 AM

```